utilities_test.go

```diff
@@ -98,84 +98,112 @@ func TestDatasetWarp(t *testing.T) {
 	}
 }
 func TestDatasetWarpMulti(t *testing.T) {
-	ds1, _ := Open("testdata/test_1_1.tif", RasterOnly())
-	ds2, _ := Open("testdata/test_1_2.tif", RasterOnly())
-	ds3, _ := Open("testdata/test_2_1.tif", RasterOnly())
-	ds4, _ := Open("testdata/test_2_2.tif", RasterOnly())
+	ds1, _ := Create(Memory, "", 3, Byte, 5, 5)
+	ds2, _ := Create(Memory, "", 3, Byte, 5, 5)
+
+	sr, _ := NewSpatialRefFromEPSG(4326)
+	_ = ds1.SetSpatialRef(sr)
+	_ = ds2.SetSpatialRef(sr)
+
+	_ = ds1.SetGeoTransform([6]float64{45, 1, 0, 35, 0, -1})
+	_ = ds2.SetGeoTransform([6]float64{50, 1, 0, 35, 0, -1})
+
+	for _, b := range ds1.Bands() {
+		_ = b.Fill(200, 0)
+	}
+
+	for _, b := range ds2.Bands() {
+		_ = b.Fill(100, 0)
+	}
 
 	defer ds1.Close()
 	defer ds2.Close()
-	defer ds3.Close()
-	defer ds4.Close()
 
 	// Warp NewDataset with multiple input dataset
 	filePath := path.Join(os.TempDir(), "warp.tif")
-	outputDataset, err := Warp(filePath, []*Dataset{ds1, ds2, ds3, ds4}, []string{"-ts", "40", "40"}, CreationOption("TILED=YES", "BLOCKXSIZE=32", "BLOCKYSIZE=16"), GTiff)
+	outputDataset, err := Warp(filePath, []*Dataset{ds1, ds2}, []string{}, CreationOption("TILED=YES"), GTiff)
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	defer os.Remove(filePath)
 	defer outputDataset.Close()
 
-	st := outputDataset.Structure()
-	if st.SizeX != 40 || st.SizeY != 40 {
-		t.Errorf("wrong size %d,%d", st.SizeX, st.SizeY)
-	}
+	data := make([]uint8, 50)
+	err = outputDataset.Read(0, 0, data, outputDataset.Structure().SizeX, outputDataset.Structure().SizeY, Bands(0, 1, 2), Window(10, 10))
+	assert.Error(t, err, "Access window out of range")
 
-	if st.BlockSizeX != 32 || st.BlockSizeY != 16 {
-		t.Errorf("wrong block size %d,%d", st.BlockSizeX, st.BlockSizeY)
-	}
+	// read total warp result
+	err = outputDataset.Read(0, 0, data, outputDataset.Structure().SizeX, outputDataset.Structure().SizeY,
+		Bands(0, 1, 2),
+		Window(outputDataset.Structure().SizeX, outputDataset.Structure().SizeY),
+	)
+	assert.NoError(t, err)
 
-	geoTransform, err := outputDataset.GeoTransform()
-	if err != nil {
-		t.Errorf("failed to get geotransform")
+	assert.Equal(t, []uint8{200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 100, 100, 100, 100, 100}, data)
```

Comment:
You can remove the tests following this one. We've checked that the size is correct and the contents is a mix of the source datasets, i.e. that gdal has been correctly inputed its sources; no need to check that the warping alg in gdal is correct.